src/material-examples/tree-nested-overview/tree-nested-overview-example.ts

```diff
@@ -110,20 +46,14 @@ export class FileDatabase {
   selector: 'tree-nested-overview-example',
   templateUrl: 'tree-nested-overview-example.html',
   styleUrls: ['tree-nested-overview-example.css'],
-  providers: [FileDatabase]
 })
 export class TreeNestedOverviewExample {
-  nestedTreeControl: NestedTreeControl<FileNode>;
-  nestedDataSource: MatTreeNestedDataSource<FileNode>;
+  treeControl: NestedTreeControl<FoodNode> = new NestedTreeControl<FoodNode>(node => node.children);
```

Comment:
The immediate typing can be removed since Typescript will infer from what you are creating